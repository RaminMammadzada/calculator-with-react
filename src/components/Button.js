import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const { buttonName, handleClick, styling } = props;
  return (
    <button
      type="button"
      onClick={() => handleClick(buttonName)}
      className={styling}
    >
      {buttonName}
    </button>
  );
};

Button.defaultProps = {
  handleClick: null,
};

Button.propTypes = {
  buttonName: PropTypes.string.isRequired,
  handleClick: PropTypes.func,
  styling: PropTypes.string.isRequired,
};

export default Button;
