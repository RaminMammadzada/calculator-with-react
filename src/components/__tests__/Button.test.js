import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '../Button';

const handleClick = () => ({ total: '0', next: '0', operation: 'x' });

describe('Button', () => {
  it('should render the Button component', () => {
    const tree = ReactTestRenderer.create(<Button handleClick={handleClick} buttonName="0" styling="zero-style" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a Button component', () => {
    render(<Button handleClick={handleClick} buttonName="0" styling="zero-style" />);
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument();
  });
});
