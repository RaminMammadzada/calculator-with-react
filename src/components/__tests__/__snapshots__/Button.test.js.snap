// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button component 1`] = `
<button
  className="zero-style"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;
